src/Utilities/CssDocument.php

```diff
@@ -110,177 +111,104 @@ public function getStyleRulesData(array $allowedMediaTypes): array
-     *   or an empty string for CSS not within a media query
-     *
-     * Example:
-     *
-     * The CSS code
-     *
-     *   "@import "file.css"; h1 { color:red; } @media { h1 {}} @media tv { h1 {}}"
-     *
-     * will be parsed into the following array:
-     *
-     *   0 => [
-     *     "css" => "h1 { color:red; }",
-     *     "media" => ""
-     *   ],
-     *   1 => [
-     *     "css" => " h1 {}",
-     *     "media" => "@media "
-     *   ]
-     *
-     * @param array<array-key, string> $allowedMediaTypes
-     *
-     * @return array<int, array{css: string, media: string}>
-     */
-    private function splitCssAndMediaQuery(array $allowedMediaTypes): array
-    {
-        $mediaTypesExpression = '';
-        if (!empty($allowedMediaTypes)) {
-            $escapedAllowedMediaTypes = \array_map(
-                static function (string $mediaType): string {
-                    return \preg_quote($mediaType, '#');
-                },
-                $allowedMediaTypes
-            );
-            $mediaTypesExpression = '|' . \implode('|', $escapedAllowedMediaTypes);
+        if ($rule instanceof CssImport) {
+            return $this->isImportRuleAllowed;
         }
 
-        $cssSplitForAllowedMediaTypes = \preg_split(
-            '#(@media\\s++(?:only\\s++)?+(?:(?=[{(])' . $mediaTypesExpression . ')' . self::MEDIA_RULE_BODY_MATCHER
-            . ')#misU',
-            $this->styleRules,
-            -1,
-            PREG_SPLIT_DELIM_CAPTURE
-        );
+        $this->isImportRuleAllowed = false;
 
-        // filter the CSS outside/between allowed @media rules
-        $cssCleaningMatchers = [
-            'import/charset directives' => '/\\s*+@(?:import|charset)\\s[^;]++;/i',
-            'remaining media enclosures' => '/\\s*+@media\\s' . self::MEDIA_RULE_BODY_MATCHER . '/isU',
-        ];
+        if (!$rule instanceof CssAtRule) {
+            return false;
+        }
 
-        $splitCss = [];
-        foreach ($cssSplitForAllowedMediaTypes as $index => $cssPart) {
-            $isMediaRule = $index % 2 !== 0;
-            if ($isMediaRule) {
-                \preg_match('/^([^{]*+){(.*)}[^}]*+$/s', $cssPart, $matches);
-                $splitCss[] = ['css' => $matches[2], 'media' => $matches[1]];
-            } else {
-                $cleanedCss = \trim(\preg_replace($cssCleaningMatchers, '', $cssPart));
-                if ($cleanedCss !== '') {
-                    $splitCss[] = ['css' => $cleanedCss, 'media' => ''];
-                }
-            }
+        switch ($rule->atRuleName()) {
+            case 'media':
+                $result = false;
+                break;
+            case 'font-face':
+                $result = $rule instanceof CssRuleSet
+                    && $rule->getRules('font-family') !== []
+                    && $rule->getRules('src') !== [];
+                break;
+            default:
+                $result = true;
+                break;
```

Comment:
The `default` case generally should not have a `break`.

Reply:
Only if it's the last case.  Some weirdos like to put the `default` first.  But we're not weird, so I've removed the redundant `break`.